### DISCUSSION TITLE
Handling Duplicate keys in json

### DISCUSSION THREAD

**Discussion by da-viper** [nlohmann/json#3392] [Q&A] [answered]
for example if i have the following json file 
```json
{
   "cat": {
      "Garfield": 10
   },
   "cat": {
      "Tom" : 11
   },
   "cat": {
      "Andy": 1
   }
```
I read that the library stores the last seen key  is there way i choose how to handle the problem , for example merging the all of the cats into on object like 
```json
{
  "cat": {
    "Garfield": 10,
    "Tom": 11,
    "Andy": 1
}
```
or selecting the first one to appear or the most common one appearing 
 
see https://github.com/nlohmann/json/discussions/2667

Thanks

**Comment by nlohmann** [comment, marked as answer]
This could be handled with a parser callback, see https://json.nlohmann.me/features/parsing/parser_callbacks/.

**Reply by nlohmann** [reply]
@Da-Viper Do you need further assistance?
